lib/viewbase.py

```diff
@@ -74,7 +76,7 @@ def redirect_kwarg(self, kw, backup=None):
         return self.redirect(to, backup)
 
     def redirect(self, to, backup=None):
-        if not is_safe_url(url=to, allowed_hosts={self.request.get_host()}):
+        if not is_safe_url(url=to, allowed_hosts={urlparse(settings.BASE_URL).hostname + ":" + str(urlparse(settings.BASE_URL).port)}):
```

Comment:
Actually, this redirect check _should_ use request.get_host(), as this is a string we are returning to the browser. We are just cheking that the browser is not going to leave the instances. Consider that the person is using url `https://aplus` to acccess `plus.cs.aalto.fi`, which is totally valid as long as they have `.cs.aalto.fi` in the dns search path.